src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,32 @@
+package duke.command;
+
+import duke.data.TaskHandler;
+import duke.storage.Storage;
+
+/**
+ * Class that represents an executable command.
+ *
+ * @author Won Ye Ji
+ */
+public abstract class Command {
+    protected TaskHandler taskHandler;
+    protected Storage storage;
+
+    /**
+     * Constructor for Command.
+     *
+     * @param th Task Handler that handles the operation.
+     * @param str Storage that holds the task information.
+     */
+    public Command(TaskHandler th, Storage str) {
+        this.taskHandler = th;
+        this.storage = str;
+    }
+
+    /**
+     * Executes the Command.
+     *
+     * @param cmd Command string to be executed.
+     */
+    public abstract void execute(String cmd);
```

Comment:
```suggestion
    public abstract void execute(String command);
```
Would it be clearer to the reader what this String represents if it was not abbreviated?